runtime/call_metrics.go

```diff
@@ -234,3 +243,54 @@ func NewOutboundTChannelMetrics(scope tally.Scope) *OutboundTChannelMetrics {
 	metrics.SystemErrors = scope.Counter(outboundCallsSystemErrors)
 	return &metrics
 }
+
+// NewOutboundCustomMetrics returns outbound metrics for custom clients
+func NewOutboundCustomMetrics(scope tally.Scope) *OutboundCustomMetrics {
+	metrics := OutboundCustomMetrics{}
+	metrics.Sent = scope.Counter(outboundCallsSent)
+	metrics.Latency = scope.Timer(outboundCallsLatency)
+	metrics.Success = scope.Counter(outboundCallsSuccess)
+	metrics.Errors = scope.Counter(outboundCallsErrors)
+	return &metrics
+}
+
+// SetupCustomClientLoggersMetrics creates:
+// - client logger/scope
+// - client method loggers/metrics
+func SetupCustomClientLoggersMetrics(
+	clientID string,
+	methods []string,
+	deps *DefaultDependencies,
+) (
+	logger *zap.Logger,
+	scope tally.Scope,
+	loggers map[string]*zap.Logger,
+	metrics map[string]*OutboundCustomMetrics,
+) {
+	logger = deps.Logger.With(zap.String("clientID", clientID))
+	scope = deps.Scope.Tagged(map[string]string{"client": clientID})
+
+	loggers = make(map[string]*zap.Logger, len(methods))
+	metrics = make(map[string]*OutboundCustomMetrics, len(methods))
+	for _, methodName := range methods {
+		loggers[methodName] = logger.With(
+			zap.String("methodName", methodName),
+		)
+		metrics[methodName] = NewOutboundCustomMetrics(scope.Tagged(map[string]string{
+			"method": methodName,
+		}))
+	}
+
+	return logger, scope, loggers, metrics
```

Comment:
why T_T

Reply:
Why what? Not sure what your question is.